lighthouse-core/audits/large-javascript-libraries.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/**
+ * @fileoverview This audit checks a page for any large JS libraries with smaller alternatives.
+ * These libraries can be replaced with functionally equivalent, smaller ones.
+ */
+
+'use strict';
+/** @typedef {{repository: string, lastScraped: number|'Error', versions: Record<string, {gzip: number}>}} BundlePhobiaLibrary */
+/** @typedef {{gzip: number, name: string, repository: string}} MinifiedBundlePhobiaLibrary */
+
+/** @type {Record<string, BundlePhobiaLibrary>} */
+const libStats = require('../lib/large-javascript-libraries/bundlephobia-database.json');
+
+/** @type {Record<string, string[]>} */
+const librarySuggestions = require('../lib/large-javascript-libraries/library-suggestions.js')
+  .suggestions;
+
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when no known unnecessarily large libraries are detected on the page.*/
+  title: 'Avoids unnecessarily large JavaScript libraries',
```

Comment:
Should we say 'Avoids large JavaScript libraries with smaller alternatives' ?

Reply:
I like that better